apps/api/src/app/subscribers/usecases/update-subscriber-channel/update-subscriber-channel.usecase.ts

```diff
@@ -118,25 +118,55 @@ export class UpdateSubscriberChannel {
       return;
     }
 
+    let deviceTokens: string[] = [];
+
+    if (updatePayload.credentials?.deviceTokens) {
+      deviceTokens = this.unionDeviceTokens(
+        existingChannel.credentials.deviceTokens ?? [],
+        updatePayload.credentials.deviceTokens
+      );
+    }
+
     await this.invalidateCache.invalidateByKey({
       key: buildSubscriberKey({
         subscriberId: foundSubscriber.subscriberId,
         _environmentId: foundSubscriber._environmentId,
       }),
     });
 
-    const mergedChannel = Object.assign(existingChannel, updatePayload);
+    const mappedChannel: IChannelSettings = this.mapChannel(updatePayload, existingChannel, deviceTokens);
 
     await this.subscriberRepository.update(
       {
         _environmentId: environmentId,
         _id: foundSubscriber,
         'channels._integrationId': existingChannel._integrationId,
       },
-      { $set: { 'channels.$': mergedChannel } }
+      { $set: { 'channels.$': mappedChannel } }
     );
   }
 
+  private mapChannel(
+    updatePayload: Partial<IChannelSettings>,
+    existingChannel: IChannelSettings,
+    deviceTokens: string[]
+  ): IChannelSettings {
+    return {
+      _integrationId: updatePayload._integrationId || existingChannel._integrationId,
+      providerId: updatePayload.providerId || existingChannel.providerId,
+      credentials: { ...existingChannel.credentials, ...updatePayload.credentials, ...{ deviceTokens } },
+    };
+  }
+
+  private unionDeviceTokens(existingDeviceTokens: string[], updateDeviceTokens: string[]): string[] {
+    // in order to not have breaking change we will support [] update
+    if (updateDeviceTokens?.length === 0) return [];
+
+    const merged = [...existingDeviceTokens, ...updateDeviceTokens];
+
+    return union(merged);
```

Comment:
Maybe instead of union we could use the set trick
` [...new Set([...existingDeviceTokens, ...updateDeviceTokens])]`

Not sure what has better performance

Reply:
actually, it should have better performance, i was not sure regarding the readability. but i tend to think that this one is more common therefore more readable, thanks for the comment ill update it :)